pynicotine/gtkgui/transfers.py

```diff
@@ -625,11 +626,25 @@ def update_specific(self, transfer):
                 parent_iterator = user_folder_path_iterator
                 user_folder_path_child_transfers.append(transfer)
 
+                if select_parent:
+                    self.tree_view.expand_row(user_iterator)
+                    select_iterator = user_folder_path_iterator
+
                 # Group by folder, path not visible in file rows
                 folder_path = ""
             else:
                 parent_iterator = user_iterator
                 user_child_transfers.append(transfer)
+
+                if select_parent:
+                    select_iterator = user_iterator
+
+            if select_iterator and (not self.tree_view.is_row_selected(select_iterator)
+                                    or self.tree_view.get_num_selected_rows() != 1):
+                # Select parent row of newly added transfer
```

Comment:
Please clarify or explain the condition.

Reply:
Done